src/conn/mod.rs

```diff
@@ -179,53 +222,65 @@ impl DNSConn {
         let raw_query = match msg.pack() {
             Ok(v) => v,
             Err(err) => {
-                warn!("Failed to construct mDNS packet {}", err);
+                log::error!("Failed to construct mDNS packet {}", err);
                 return;
             }
         };
 
-        trace!("{:?} sending {:?}...", self.socket.local_addr(), raw_query);
+        log::trace!("{:?} sending {:?}...", self.socket.local_addr(), raw_query);
         if let Err(err) = self.socket.send_to(&raw_query, self.dst_addr).await {
-            warn!("Failed to send mDNS packet {}", err);
+            log::error!("Failed to send mDNS packet {}", err);
         }
     }
 
     async fn start(
         mut closed_rx: mpsc::Receiver<()>,
+        close_server: Arc<atomic::AtomicBool>,
         socket: Arc<UdpSocket>,
         local_names: Vec<String>,
         dst_addr: SocketAddr,
         queries: Arc<Mutex<Vec<Query>>>,
     ) -> Result<(), Error> {
-        let mut b = vec![0u8; INBOUND_BUFFER_SIZE];
-
-        let (mut n, mut src);
+        log::info!("enter loop and listening {:?}", socket.local_addr());
 
         loop {
-            trace!("enter loop and listening {:?}", socket.local_addr());
+            let mut b = vec![0u8; INBOUND_BUFFER_SIZE];
```

Comment:
Will there be any performance issue to allocate vector b inside loop ? Is this change necessary?

Reply:
None whatsoever. Fixed changes.